terraform/environments/oas/backups.tf

```diff
@@ -0,0 +1,65 @@
+locals {
+  cold_storage_after = 7
+}
+
+resource "aws_backup_vault" "default_oas" {
+  name =  "${local.application_name}-backup-vault"
+  tags = merge(
+    local.tags,
+    { "Name" = "${local.application_name}-backup-vault" },
+  )
+}
+
+# Non production backups
+resource "aws_backup_plan" "non_production_oas" {
+
+  name = "${local.application_name}-backup-daily-retain-7-days"
+
+  rule {
+    rule_name         = "${local.application_name}-backup-daily-retain-7-days"
+    target_vault_name = aws_backup_vault.default_oas.name
+
```

Comment:
Some of these options can be turned into local variables for ease of changing in the future